packages/csvviewer/src/widget.ts

```diff
@@ -99,36 +97,23 @@ class CSVViewer extends Widget {
   }
 
   /**
-   * The CSV data model.
-   */
-  get model(): CSVModel {
-    return this._model;
-  }
-
-  /**
    * Dispose of the resources used by the widget.
    */
   dispose(): void {
-    if (this._model === null) {
+    if (this._grid === null) {
       return;
     }
-    let model = this._model;
-    let table = this._table;
+    let grid = this._grid;
     let toolbar = this._toolbar;
-    let warning = this._warning;
     let monitor = this._monitor;
-    this._model = null;
-    this._table = null;
+    this._grid = null;
     this._toolbar = null;
-    this._warning = null;
     this._monitor = null;
 
-    model.dispose();
-    table.dispose();
+    grid.dispose();
```

Comment:
You don't need to dispose of your child widgets. The layout does it automatically.